### ISSUE TITLE
[A11Y] Properly indicate the "error" status of a form input and point to the related error message

### ISSUE THREAD

**Issue by indigoxela** [backdrop/backdrop-issues#5672]
## Description of the bug

Currently, if validation of a form item fails, there's backdrop_messages() (usually somewhere above) and a visual indicator on the form item, based on a css class.

That's however not sufficient re A11Y, we should also consider:

1. Using aria-invalid (will cause the screen reader to announce "invalid" when that control gets the focus)
2. Using aria-errormessage (to properly connect failing form items with their error messages)

## Additional info

- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-invalid
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage

Note that there are existing issue, that might make things easier here:

- #5348
- #1040
